### PR TITLE
Enhancement - PokemonsToTransferImmediately 

### PR DESCRIPTION
**Pull request by rayn21dev**
Enhancement - PokemonsToTransferImmediately - Catch and transfer insead of skipping. 

#613

### CODE REVIEW

File: PoGo.NecroBot.Logic/Tasks/CatchPokemonTask.cs [hunk 1]
```diff
@@ -104,7 +104,41 @@ public static void Execute(Context ctx, StateMachine machine, EncounterResponse
...
+                        ctx.Inventory.DeletePokemonFromInvById(encounter.WildPokemon.PokemonData.Id);
+
+                        var bestPokemonOfType = ctx.LogicSettings.PrioritizeIvOverCp
+                            ? ctx.Inventory.GetHighestPokemonOfTypeByIv(encounter.WildPokemon.PokemonData).Result
+                            : ctx.Inventory.GetHighestPokemonOfTypeByCp(encounter.WildPokemon.PokemonData).Result;
+
+                        var pokemonSettings = ctx.Inventory.GetPokemonSettings().Result;
+                        var pokemonFamilies = ctx.Inventory.GetPokemonFamilies().Result;
+
+                        var setting = pokemonSettings.Single(x => x.PokemonId == pokemon.PokemonId);
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Tasks/CatchPokemonTask.cs]: This code was crashing, look at the fix I added

**Comment by Nordiii** [comment]
Would remove the pinsir as default!

File: PoGo.NecroBot.CLI/Settings.cs [hunk 2]
```diff
@@ -204,12 +204,18 @@ public void Save(string path)
                     PokemonId.Mewtwo
                 };
 
-        public List<PokemonId> PokemonsToEvolve = new List<PokemonId>
+        public List<PokemonId> PokemonsToImmediatelyTransfer = new List<PokemonId>
+                {
+                    PokemonId.Pinsir
```
Comment (Nordiii) [inline PoGo.NecroBot.CLI/Settings.cs]: Would remove it as default

**Comment by Wil32** [comment]
Can you execute this after evolve?
ex. Golbat?

**Comment by rayn21dev** [comment]
@Wil32 I did not implement it after an evolve.  I will see about adding that this evening.

@Nordiii I will remove Pinsir as the default.

**Comment by rayn21dev** [comment]
@yamashi I don't see the fix you mention - what am I missing?

**Comment by lorenzowalz** [comment]
@rayn21dev 

Any interest in using a config file for the PokemonsToImmediatelyTransfer? That seems to be the way this project is going to allow a lot of optimization / min maxing.

**Comment by rayn21dev** [comment]
@lorenzowalz I think you can use the config file to do this, just like the other lists (I am testing)

**Comment by Artud** [comment]
Fix conflicts.

**Pull request closed without merging**